src/main/java/com/skytask/listener/Listener.java

```diff
@@ -0,0 +1,39 @@
+package com.skytask.listener;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.skytask.channel.ProductSource;
+import com.skytask.model.Product;
+import com.skytask.service.ProductService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.messaging.handler.annotation.SendTo;
+
+import java.util.List;
+
+@EnableBinding(ProductSource.class)
+public class Listener {
+
+    @Autowired
+    private ProductService productService;
```

Comment:
Why are you autowiring the value this way instead of doing it on the constructor?